tests/test_subscriber.py

```diff
@@ -193,19 +195,114 @@ def test_on_put_invalid(self, mock_invalid_param):
         self.assertTrue('invalid_param' in str(context.exception))
 
 
+@patch.object(Persister, 'create_indexes')
+@patch.object(Config, 'load_defaults')
+@patch('history.subscriber.persister.Messenger')
+def test_persist_all_events(mock_messenger, mock_config, mock_create_indexes):
+
+    from history.subscriber.persister import start_dojot_messenger
+    from history import conf
+
+    p = Persister()
+    p.create_indexes_for_notifications('admin')
+
+    mock_config.dojot = {
+        "management": {
+            "user": "dojot-management",
+            "tenant": "dojot-management"
+        },
+        "subjects": {
+            "tenancy": "dojot.tenancy",
+            "devices": "dojot.device-manager.device",
+            "device_data": "device-data"
+        }
+    }
+
+    # test persist all boolean valued events
+    start_dojot_messenger(mock_config, p, False)
+    mock_messenger.assert_called_once_with("Persister", mock_config)
+
+    mock_messenger().create_channel.assert_any_call("dojot.notifications", "r")
+
+    mock_messenger().on.assert_any_call(mock_config.dojot['subjects']['tenancy'],"message", p.handle_new_tenant)
+
+    mock_messenger().on.assert_any_call("dojot.notifications", "message",p.handle_notification)
+
+    mock_messenger().create_channel.assert_any_call(mock_config.dojot['subjects']['devices'], "r")
+
+    mock_messenger().create_channel.assert_any_call(mock_config.dojot['subjects']['device_data'], "r")
+
+    mock_messenger().on.assert_any_call(mock_config.dojot['subjects']['devices'],"message", p.handle_event_devices)
+
+    mock_messenger().on.assert_any_call(mock_config.dojot['subjects']['device_data'],"message", p.handle_event_data)
+
+
+@patch.object(Persister, 'create_indexes')
+@patch.object(Config, 'load_defaults')
+@patch('history.subscriber.persister.Messenger')
+def test_persist_only_notifications(mock_messenger, mock_config, create_indexes):
+
+    from history.subscriber.persister import start_dojot_messenger
+    from history import conf
+    p = Persister()
+    p.create_indexes_for_notifications('admin')
+
+    mock_config.dojot = {
+        "management": {
+            "user": "dojot-management",
+            "tenant": "dojot-management"
+        },
+        "subjects": {
+            "tenancy": "dojot.tenancy",
+            "devices": "dojot.device-manager.device",
+            "device_data": "device-data"
+        }
+    }
+    # test persist only boolean valued notifications
+    start_dojot_messenger(mock_config, p, True)
```

Comment:
Eu acho que eu havia te pedido para verificar se as coisas NÃO foram chamadas também... Veja os comentários no outro PR.

Reply:
Here you can do something like, check if messenger.create_channel was called with the parameters "dojot.notifications", "r" and and called only once

You could also check that the calls in messenger.on are called properly with the expected parameters.

And if messenger.create_channel is called 1 times and messenger.on is called 2 times